Library/Homebrew/cmd/update-bash.sh

```diff
@@ -2,22 +2,41 @@ brew() {
   "$HOMEBREW_BREW_FILE" "$@"
 }
 
+git() {
+  [[ -z "$HOMEBREW_GIT" ]] && HOMEBREW_GIT="$(which_git 2>/dev/null)"
```

Comment:
Yeh, I get that, I just think it's a pretty major change that should be handled in another PR along with e.g. a trap to improve the messaging.